akka-stream/src/main/scala/akka/stream/impl/fusing/GraphStages.scala

```diff
@@ -433,7 +433,12 @@ import scala.concurrent.{ Future, Promise }
           promise.tryFailure(ex)
         }
 
+        override def postStop(): Unit = {
+          if (!promise.isCompleted) promise.tryFailure(new AbruptStageTerminationException(this))
```

Comment:
the if is not needed I'd say

Reply:
Just wanted to avoid creating the exception in the normal case.

Reply:
Yeah good reason actually, LGTM :)